UI/output-obj.cpp

```diff
@@ -0,0 +1,149 @@
+#include "output-obj.hpp"
+#include "obs-frontend-api.h"
+
+static void onEvent(enum obs_frontend_event event, void *param)
+{
+	OutputObj *obj = static_cast<OutputObj *>(param);
+
+	OBSSourceAutoRelease source;
+
+	switch (event) {
+	case OBS_FRONTEND_EVENT_STUDIO_MODE_ENABLED:
+	case OBS_FRONTEND_EVENT_PREVIEW_SCENE_CHANGED:
+		source = obs_frontend_get_current_preview_scene();
+		break;
+	case OBS_FRONTEND_EVENT_STUDIO_MODE_DISABLED:
+		source = obs_frontend_get_current_scene();
+		break;
+	case OBS_FRONTEND_EVENT_SCENE_CHANGED:
+		if (!obs_frontend_preview_program_mode_active())
+			source = obs_frontend_get_current_scene();
+
+		break;
+	default:
+		break;
+	}
+
+	if (source)
```

Comment:
I think it would be best to remove this check altogether. If a scene is changed to nullptr, it's a good time to release the current source anyway.

Reply:
Fixed